code/game/objects/items/manuals.dm

```diff
@@ -235,15 +235,20 @@
 
 // Wiki books that are linked to the configured wiki link.
 
+#define BOOK_WINDOW_BROWSE_SIZE "970x710"
+
 // A book that links to the wiki
 /obj/item/book/manual/wiki
+	/// The ending URL of the page that we link to.
 	var/page_link = ""
-	window_size = "970x710"
 
-/obj/item/book/manual/wiki/attack_self()
-	if(!book_data.content)
+/obj/item/book/manual/wiki/display_content(mob/living/user)
+	if(isnull(book_data.content))
 		initialize_wikibook()
-	return ..()
+
+	credit_book_to_reader(user)
+	var/book_content = book_data.get_content(trusted = TRUE) // legitimately one of the only times you should be using this argument
```

Comment:
dont make it an argument make it another proc it's too dangerous to be left alive

Reply:
and add a lint for existing only being called once

Reply:
Maybe I should just access the content variable directly for this proc

Reply:
even better i just made it a macro

Reply:
yeah this is good.